tests/query.rs

```diff
@@ -882,6 +882,58 @@ fn test_basic_aggregates() {
     }
 }
 
+#[test]
+fn test_combinatorial() {
+    let mut store = Store::open("").expect("opened");
+
+    store.transact(r#"[
+        [:db/add "a" :db/ident :foo/name]
+        [:db/add "a" :db/valueType :db.type/string]
+        [:db/add "a" :db/cardinality :db.cardinality/one]
+        [:db/add "b" :db/ident :foo/dance]
+        [:db/add "b" :db/valueType :db.type/ref]
+        [:db/add "b" :db/cardinality :db.cardinality/many]
+        [:db/add "b" :db/index true]
+    ]"#).unwrap();
+
+    store.transact(r#"[
+        [:db/add "a" :foo/name "Alice"]
+        [:db/add "b" :foo/name "Beli"]
+        [:db/add "c" :foo/name "Carlos"]
+        [:db/add "d" :foo/name "Diana"]
+
+        ;; Alice danced with Beli.
+        [:db/add "a"  :foo/dance "ab"]
+        [:db/add "b"  :foo/dance "ab"]
```

Comment:
what would happen if I did `[:db/add "b" :foo/dance "ba"]` here? Does this rely upon the 'dance' ordering being identical for unpermute to work?

Reply:
That would add a second dance for Beli, one in which she doesn't dance with anyone else. The only set of bindings for `?left` and `?right` would be her, and naturally she doesn't unpermute with herself, so that solo dance wouldn't contribute anything to the output.

If you were to add both Alice and Beli to that second dance, the output would be unchanged: the inner query would generate `[Alice, Beli] [Alice, Alice] [Beli, Alice] [Beli, Beli]` _twice_, once for each of the two dances; the `unpermute` step would remove `[x, x]` pairs and one half of the orders; and then the internal `DISTINCT` would remove duplicates.

I've added this to the test.